resources/backend/api_endpoints.md

```diff
@@ -0,0 +1,35 @@
+# API Endpoints
+API endpoints are essentially points of communication in which a user can interact with and utilize an API's functions. In the context of this project, they are ways for frontend users to interact with and use backend functions.
+![API Endpoint Diagram](../images/api_endpoint_diagram.png)
+Imagine the backend is an island with multiple ship ports. If a frontend user wants to interact with the backend, they send a ship to one of the ports to either deliver, retrieve, or modify information. These ships are called **HTTP requests**.
+## What does an endpoint look like?
+API endpoints are identified with a URL very similar to website URLs (e.g. `http://andymathnavigator.com/api/programs/v1/...`). The specific syntax of the API URL depends on how the backend developers designed it.
+
+To demonstrate, the programs domain of this project will be used as an example.
+
+API endpoints can receive URI parameters. For example, if you want to access a program with id "prog1", the syntax for this specific API is `/api/programs/v1/program/prog1`. If you want to access the program with id "prog2", send a HTTP request to `/api/programs/v1/program/prog2`.
+
+API endpoints can also receive query parameters. For example, if you want to access all published programs from this sample API, you would use `/api/programs/v1/all?published=true`. You've probably seen query parameters in website URLs before. For example, if you search something in Google, the URL will look something like `https://www.google.com/search?source=hp&ei=DFKbXvqAIPCIytMP...`. The stuff after the question mark is a query parameter.
+
+We use URI parameters when we want to identify a specific resource(s), and we use query parameters as to sort or filter resources.
+## What does an HTTP request look like?
+HTTP requests have multiple types, known as methods. A GET request retrieves information from an API, a POST request sends information to an API, and a DELETE request removes information from an API. There are other methods, but this project doesn't use them (yet).
+
```

Comment:
I would do a bullet point for this: The main types of HTTP requests are:
 - GET - retrieves ....
 - POST - ....
 - DELETE - ....

With explanations after each type. Or something like that?